src/rust/ensogl/lib/components/src/drop_down_menu.rs

```diff
@@ -298,26 +305,16 @@ impl DropDownMenu {
             // === Selection ===
 
             eval_ model.selection_menu.chosen_entry (hide_menu.emit(()));
-            chosen_entry_unmasked <- model.selection_menu.chosen_entry.map(f!((entry_id)
-                model.get_unmasked_index(*entry_id))
-            );
-            frp.source.chosen_entry <+ chosen_entry_unmasked;
-            set_selected            <- any(frp.input.set_selected, chosen_entry_unmasked);
+            frp.source.chosen_entry <+ model.selection_menu.chosen_entry;
+
+            set_selected <- any(frp.input.set_selected,model.selection_menu.chosen_entry);
 
             eval set_selected([model](entry_id) {
                 if let Some(entry_id) = entry_id {
-                    if let Some(content) = model.content.borrow().as_ref() {
-                        // We get an external item index, so we operate on all items, thus we
-                        // clear the mask.
-                        content.clear_mask();
-                        if let Some(item) = model.get_content_item(Some(*entry_id)) {
-                            model.set_label(&item.label)
-                        };
-                        // Remove selected item from menu list
-                        content.set_mask(*entry_id);
-                        // Update menu content.
-                        let entries:list_view::entry::AnyModelProvider = content.clone().into();
-                        model.selection_menu.frp.set_entries.emit(entries);
+                    if let Some(_content) = model.content.borrow().as_ref() {
+                        if let Some(item) = model.get_item_label(Some(*entry_id)) {
+                            model.set_label(item.as_ref())
+                        }
```

Comment:
This code is only needed if we use the `MaskedModelProvider`. I removed it because it was not working and it did not seem worth it to adapt broken code to the new list view interface.

Reply:
If I understand correctly, MaskedModelProvider was working (at least partially) as it was hiding the currently selected visualization in the visualization chooser. I understand that this PR introduces regression there, is this correct? I mean, in visualization chooser we need to hide the currently selected entry and it was working, so it had to work somehow. Anyway, removing such things should be done by first talking about it with people who implemented it. Have you conducted such talks with Adam and did he agree for it to be removed? If not, this should be reverted / fixed and discussed properly.